contracts/InterchainTokenService.sol

```diff
@@ -306,6 +306,34 @@ contract InterchainTokenService is
         );
     }
 
+    /**
+     * @notice Only to be used by the InterchainTokenFactory to register custom tokens to this chain. Then link token can be used to register those tokens to other chains.
+     * @param salt a unique salt to derive tokenId from.
+     * @param tokenManagerType the type of the token manager to use for the token registration.
+     * @param linkParams the operator for the token.
+     */
+    function registerCustomToken(
+        bytes32 salt,
+        address tokenAddress,
+        TokenManagerType tokenManagerType,
+        bytes calldata linkParams
+    ) external payable whenNotPaused returns (bytes32 tokenId) {
+        // Custom token managers can't be deployed with native interchain token type, which is reserved for interchain tokens
+        if (tokenManagerType == TokenManagerType.NATIVE_INTERCHAIN_TOKEN) revert CannotDeploy(tokenManagerType);
+
+        if ( msg.sender != interchainTokenFactory ) {
+            revert NotInterchainTokenFactory(msg.sender);
+        }
+
+        address deployer = TOKEN_FACTORY_DEPLOYER;
```

Comment:
```suggestion
        if (msg.sender != interchainTokenFactory) {
            revert NotInterchainTokenFactory(msg.sender);
        }

        address deployer = TOKEN_FACTORY_DEPLOYER;

        // Custom token managers can't be deployed with native interchain token type, which is reserved for interchain tokens
        if (tokenManagerType == TokenManagerType.NATIVE_INTERCHAIN_TOKEN) revert CannotDeploy(tokenManagerType);
```

Reply:
add an `onlyFactory` modifier perhaps

Reply:
i thought about it but it is not reused anywhere.